editor/editor_interface.cpp

```diff
@@ -263,6 +265,85 @@ void EditorInterface::set_current_feature_profile(const String &p_profile_name)
 	EditorFeatureProfileManager::get_singleton()->set_current_profile(p_profile_name, true);
 }
 
+// Editor dialogs.
+
+void EditorInterface::popup_dialog_scene_tree(const Callable &p_callback, const TypedArray<StringName> &p_valid_types) {
+	if (dialog_scene_tree) {
+		dialog_scene_tree->queue_free();
+	}
+	dialog_scene_tree = memnew(SceneTreeDialog);
+
+	Vector<StringName> valid_types;
+	int length = p_valid_types.size();
+	valid_types.resize(length);
+	for (int i = 0; i < length; i++) {
+		valid_types.write[i] = p_valid_types[i];
+	}
+	dialog_scene_tree->set_valid_types(valid_types);
+
+	get_base_control()->add_child(dialog_scene_tree);
+
+	dialog_scene_tree->popup_scenetree_dialog();
+
+	const Callable selected_callback = callable_mp(this, &EditorInterface::_node_selected).bind(p_callback);
+	dialog_scene_tree->connect(SNAME("selected"), selected_callback, CONNECT_DEFERRED);
+
+	const Callable canceled_callback = callable_mp(this, &EditorInterface::_node_selection_canceled).bind(p_callback);
+	dialog_scene_tree->connect(SNAME("canceled"), canceled_callback, CONNECT_DEFERRED);
+}
+
+void EditorInterface::popup_dialog_property_selector(Object *p_object, const Callable &p_callback, const Vector<int> &p_type_filter) {
+	if (dialog_property_selector) {
+		dialog_property_selector->queue_free();
+	}
+	dialog_property_selector = memnew(PropertySelector);
+
+	Vector<Variant::Type> type_filter;
+	int length = p_type_filter.size();
+	type_filter.resize(length);
+	for (int i = 0; i < length; i++) {
+		type_filter.write[i] = (Variant::Type)p_type_filter[i];
+	}
+	dialog_property_selector->set_type_filter(type_filter);
+
+	get_base_control()->add_child(dialog_property_selector);
+
+	dialog_property_selector->select_property_from_instance(p_object);
+
+	const Callable selected_callback = callable_mp(this, &EditorInterface::_property_selected).bind(p_callback);
```

Comment:
You could connect the dialog once and store callback in the class' property instead.

Reply:
What would the advantage of that be?